packages/vite/src/node/utils.ts

```diff
@@ -1204,3 +1204,9 @@ export function joinUrlSegments(a: string, b: string): string {
   }
   return a + b
 }
+
+export function stripBase(path: string, base: string): string {
+  const devBase = base.endsWith('/') ? base.slice(0, -1) : base
+
+  return path.replace(RegExp(devBase + '/+'), '/')
```

Comment:
I think the following would be equivalent and just slightly simpler
```suggestion
  const devBase = base.endsWith('/') ? base : base + '/'
  return path.replace(RegExp(devBase + '+'), '/')
```

Reply:
what do we need the `+` for at the end of the regex? can we omit that?

Reply:
Yeah, somehow it still works - probably because the case `/foo` is handled without calling `stripBase()` anyway. But, what's the intended behavior when calling `stripBase("/foobar", "/foo")`? Should that be reduced to "bar" (without any slash)?

I think we need to discuss/decide, what "base" actually means, e.g., is that "any string-prefix" or is that a "string covered in slashes"? I think right now it's the later and we only want to change one special case, i.e., when calling the set base, but leaving the trailing slash.

In the future, do we want to have `/foo` and `/foo/` being handled complete the same or should there be any difference between them?

| URL       | base="/foo/"                     | base="/foo" |
|-----------|----------------------------------|-------------|
| /foo      | 404, did you mean "/foo/"?*      | Root-Path |
| /foo/     | Root-Path | Root-Path |
| /foobar   | 404, did you mean "/foo/foobar"? | ???         |
| /foo/bar| "bar"-Path | "bar"-Path |
| /foo/bar/ | "bar"-Path | "bar"-Path |

\* Right now it says "did you mean "/foo/foo"

Reply:
good point. another issue is making sure we're only replacing the start of the string. it should probably look something more like this:
```
export function stripBase(path: string, base: string): string {
  if (path === base) {
    return '/';
  }
  const devBase = base.endsWith('/') ? base : base + '/'
  return path.replace(RegExp('^' + devBase), '/')
}
```